src/statsmodel.jl

```diff
@@ -45,12 +45,15 @@ struct DataFrameRegressionModel{M,T} <: RegressionModel
     mm::ModelMatrix{T}
 end
 
+# Overload this function for a specific model type, according to whether it requires an intercept
```

Comment:
Can you turn this comment into a docstring, and develop it a little bit more (e.g. what function uses it, and what does it do exactly when it is set)?

Reply:
Sure. I'm also not 100% sure about the name. Now it defaults to `true` and only is effective when set to `false`, in which case it removes the intercept, so maybe it'd be clearer to instead define `exclude_intercept` defaulting to `false`. What do you think?

Reply:
"allows" sounds like it would throw an error if it was requested. Maybe `include_intercept` or `use_intercept`? Or `supports_intercept`, which refers more directly to the model type?

Reply:
I like `supports_intercept`. Should the `ModelFrame` keyword be the same? I.e. `ModelFrame(f, df, contrasts=contrasts, supports_intercept=supports_intercept(T))`

Reply:
OK. I'd keep `intercept::Bool` for `ModelFrame`, as you really decide whether you want an intercept or not in that case (it could even be used for models which support intercepts).

Reply:
Another option would be to not add any keyword to ModelFrame and simply pass it the modified `Terms` (already without intercept)

Reply:
If we keep an `intercept` keyword, with an eye on all models, I'd be in favor of not restricting to Bool but allow:
- `intercept = :auto` -> For now, doesn't do anything, in the future it will resorts to the defaults for the given Model (unless explicitly provided by the use) and error if the user didn't provide anything 
- `intercept = true` -> Adds the intercept
- `intercept = false` -> Removes the intercept

As it is in this PR the asymmetry may be confusing (`true` doesn't do anything and `false` removes the intercept). Right now `ModelFrame(@formula(y ~x), df, intercept = true)` would actually have no intercept.

Reply:
A boolean would make sense if we call it `supports_intercept`: then the asymmetry would be OK. But it's weird to have this argument to `ModelFrame`.

Modifying the `Terms` object sounds simpler. I guess @kleinschmidt will have an opinion, but AFAICT better apply the smallest possible change, at least for now.

Reply:
Sure, I can update the PR as soon as a consensus is reached. The `Terms` object approach seems the simplest and most sensible to me (and the one with the smallest diff).